### PR TITLE
feat: add datadog_tracing directive

### PR DESCRIPTION
**Pull request by dubloom**
- Deprecation for datadog_enable / datadog_disable
- Creation of datadog_tracing on/off (which do exactly the same as datadog_enable / datadog_disable)
- Creation of multiple macro for deprecating directive (one for 1.2.0, one for 1.4.0)
- Add a new test to see if the "specialization" of datadog_tracing on/off was working

### CODE REVIEW

File: test/cases/auto_propagation/conf/http_disabled_at_http_enabled_at_location.conf
[the remark this answers is not in the document]
Comment (dubloom) [reply test/cases/auto_propagation/conf/http_disabled_at_http_enabled_at_location.conf:1]: Done

File: src/ngx_http_datadog_module.cpp [hunk 1]
```diff
@@ -60,10 +60,16 @@ using namespace datadog::nginx;
...
+#define DEFINE_DEPRECATED_COMMAND_1_2_0(NAME, TYPE)        \
+  {                                                        \
+    ngx_string(NAME), TYPE, warn_deprecated_command_1_2_0, \
+        NGX_HTTP_LOC_CONF_OFFSET, 0, NULL                  \
+  }
+
+#define DEFINE_DEPRECATED_COMMAND_1_4_0(NAME, TYPE)        \
+  {                                                        \
+    ngx_string(NAME), TYPE, warn_deprecated_command_1_4_0, \
+        NGX_HTTP_LOC_CONF_OFFSET, 0, NULL                  \
```
[the remark this answers is not in the document]
Comment (dubloom) [reply src/ngx_http_datadog_module.cpp:72]: We cannot parametrize the function called by the directive. So it is hard to parametrize this.

File: src/datadog_directive.cpp [hunk 2]
```diff
@@ -548,6 +532,19 @@ char *warn_deprecated_command(ngx_conf_t *cf, ngx_command_t * /*command*/,
...
+  assert(cf->args->nelts >= 1);
+
+  ngx_log_error(
+      NGX_LOG_WARN, cf->log, 0,
+      "Directive \"%V\" is deprecated and can be removed since v1.4.0.",
+      &elements[0]);
+
+  return NGX_OK;
+}
+
```
[the remark this answers is not in the document]
Comment (dubloom) [reply src/datadog_directive.cpp:547]: Because of the commentary below, i don't think it is possible

**Comment by dubloom** [comment]
> I don't know if I agree with us completely deprecating commands straight away without keeping backwards compatibility at least for a while. Or if we do, let's do it completely without showing a deprecation message that doesn't really explain how to solve the problem. We should define a strategy on how to approach deprecations

We can also create more specific deprecation function. I.e a function which will print: it is deprecated, use this new directive instead

**Review by pablomartinezbernardo — COMMENTED** [review]
> I.e a function which will print: it is deprecated, use this new directive instead

Let's do that where applicable!

File: doc/API.md [hunk 3]
```diff
@@ -190,32 +190,23 @@ Sampling delegation is `off` by default. The directive's argument can be a
 variable expression that evaluates to either of `on` or `off`. If the
 directive's argument is omitted, then it is as if it were `on`.
 
-### `datadog_enable`
-- **syntax** `datadog_enable`
+### `datadog_tracing`
+- **syntax** `datadog_tracing [on|off]`
```
Comment (dmehala) [inline doc/API.md:194]:
```suggestion
- **syntax** `datadog_tracing on|off`
```

**Review by dmehala — APPROVED** [review]
LGTM! :shipit: Good work! :)

**Comment by dubloom** [comment]
I resolve the conflicts, can you give a quick look please @pablomartinezbernardo ?

**Review by pablomartinezbernardo — APPROVED** [review]

**Pull request closed without merging**